test/functional/feature_assumeutxo.py

```diff
@@ -204,6 +202,40 @@ def test_snapshot_with_less_work(self, dump_output_path):
             assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", node.loadtxoutset, dump_output_path)
         self.restart_node(0, extra_args=self.extra_args[0])
 
+    def test_snapshot_in_a_divergent_chain(self, dump_output_path):
+        n0 = self.nodes[0]
+        n3 = self.nodes[3]
+        n4 = self.nodes[4]
+        assert_equal(n0.getblockcount(), FINAL_HEIGHT)
+        assert_equal(n3.getblockcount(), START_HEIGHT)
+        assert_equal(n4.getblockcount(), START_HEIGHT)
+
+        self.log.info(f"Check importing a snapshot where current chain-tip is not an ancestor of the snapshot block but has less work")
+        # Generate a divergent chain in n3 up to 298
+        self.generate(n3, nblocks=99, sync_fun=self.no_op)
+        assert_equal(n3.getblockcount(), SNAPSHOT_BASE_HEIGHT - 1)
+
+        # Try importing the snapshot and assert its success
+        self.log.info('Importing the snapshot into n3')
+        loaded = n3.loadtxoutset(dump_output_path)
+        assert_equal(loaded['coins_loaded'], SNAPSHOT_BASE_HEIGHT)
```

Comment:
I would remove this line because it doesn't really add anything that the line below doesn't test.

Reply:
Maybe rather check the chainstates instead if you want to add something more but I think it's not absolutely necessary.

Reply:
Thanks. I removed the line as suggested. I wasn't sure which chainstates check might be useful to add. Something like [this](https://github.com/bitcoin/bitcoin/blob/master/test/functional/feature_assumeutxo.py#L322C7-L328C51), maybe?

Reply:
Yeah, just that each of them is at the height you expect.

Reply:
Thanks. Updated